fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/PrefDownloadLanguages.kt

```diff
@@ -0,0 +1,513 @@
+                learnMoreText,
+            ),
+            linkTextStates = listOf(learnMoreState),
+            style = FirefoxTheme.typography.subtitle1.copy(
+                color = FirefoxTheme.colors.textPrimary,
+            ),
+            linkTextDecoration = TextDecoration.Underline,
+        )
+    }
+}
+
+@Composable
+private fun prefDownloadLanguageItemContentDescription(
+    item: PrefDownloadLanguageItem,
+    itemDescription: String,
+): String {
+    val contentDescription: String
+
+    when (item.state.status) {
+        PrefDownloadLanguageItemStatus.Downloaded -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription
+        }
+
+        PrefDownloadLanguageItemStatus.NotDownloaded -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription
+        }
+
+        PrefDownloadLanguageItemStatus.DownloadInProgress -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription + " " + stringResource(
+                    id = R.string.translations_bottom_sheet_translating_in_progress,
+                )
+        }
+
+        PrefDownloadLanguageItemStatus.Selected -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription
+        }
+    }
+    return contentDescription
+}
+
+@Composable
+private fun PrefIconDownloadLanguageItem(item: PrefDownloadLanguageItem) {
+    when (item.state.status) {
+        PrefDownloadLanguageItemStatus.Downloaded -> {
+            if (item.state.type != PrefDownloadLanguageItemType.EnglishLanguage) {
+                Icon(
+                    painter = painterResource(
+                        id = R.drawable.ic_delete,
+                    ),
+                    contentDescription = null,
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+        }
+
+        PrefDownloadLanguageItemStatus.NotDownloaded -> {
+            if (item.state.type != PrefDownloadLanguageItemType.EnglishLanguage) {
+                Icon(
+                    painter = painterResource(
+                        id = R.drawable.ic_download,
+                    ),
+                    contentDescription = null,
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+        }
+
+        PrefDownloadLanguageItemStatus.DownloadInProgress -> {
+            if (item.state.type != PrefDownloadLanguageItemType.EnglishLanguage) {
+                DownloadIconIndicator(
+                    icon = painterResource(id = R.drawable.mozac_ic_sync_24),
+                )
+            }
+        }
+
+        PrefDownloadLanguageItemStatus.Selected -> {
+            Icon(
+                painter = painterResource(
+                    id = R.drawable.mozac_ic_checkmark_24,
```

Comment:
I think the checkmark was removed from the Figma, please let me know if there is a version with it still.

Reply:
<img width="439" alt="Screenshot 2023-12-04 at 16 20 36" src="https://github.com/mozilla-mobile/firefox-android/assets/8118371/0120503c-2a81-4570-b46b-b2dce1d169af">